network/src/protocols/tests/mod.rs

```diff
@@ -312,8 +312,16 @@ fn test_identify_behavior() {
     wait_connect_state(&node1, 1);
     wait_connect_state(&node2, 0);
 
+    if !wait_until(10, || {
+        node1.network_state.with_peer_store_mut(|peer_store| {
+            peer_store.is_addr_banned(&node2.listen_addr)
+                && peer_store.addr_manager().get(&node2.listen_addr).is_none()
+        })
+    }) {
+        panic!("identify can't ban not same net")
+    }
+
     let sessions = node3.connected_sessions();
-    assert_eq!(sessions.len(), 1);
```

Comment:
Why remove this assertion?

Reply:
Because the assert is wrong here, the `registered` behavior is before `received`, and the `discovery` in the test is broadcast every second. It is impossible to determine whether the node3 is just in the process of trying to connect when asserting. It can also be changed to `wait_connect_state(&node3, 1)`, but I don’t think it’s necessary, it’s no problem to delete it directly

Reply:
According to the PR description, this PR fixes the failure of https://github.com/nervosnetwork/ckb/runs/3673717980#step:4:1915

As far as I know, this failure is due to https://github.com/nervosnetwork/ckb/pull/3052#discussion_r715374379

Then, it makes me confusing that I cannot figure out the relationship between this testing failure and "banned".